encrypted-media-respec.html

```diff
@@ -1876,8 +1886,18 @@
           <p>The following steps are run:</p>
           <ol>
             <li><p>Let the <var>session</var> be the associated <a>MediaKeySession</a> object.</p></li>
+            <li>
+              <p>
+                Destroy any license and/or keys associated with <var>session</var> that have not been explicitly stored.
+              </p>
+            </li>
             <li><p>Run the <a def-id="update-key-statuses-algorithm"></a> algorithm on the <var>session</var>, providing an empty sequence.</p></li>
             <li><p>Run the <a def-id="update-expiration-algorithm"></a> algorithm on the <var>session</var>, providing <code>NaN</code>.</p></li>
+            <li>
+              <p>
+                If the object's <var>session type</var> is <a def-id="persistent-usage-record-session"></a>, store the <var>record of key usage</var>.
```

Comment:
On the order of messages / storing data for both remove() and Session Close, this order is not observable, but I agree that it is more readable to have the storage of the *record of key usage* or the *record of license destruction* described next to the destruction of the keys or license respectively. I'll create a separate issue for that.

On the second point, yes, the *record of key usage* or *record of license destruction* should be stored only if keys or licenses were destroyed at that time. Perhaps both the destruction and messaging should be in an "if any license / keys exist..." ? This would also address your first point.